File: 278-first-bad-version/278-first-bad-version.java
/* The isBadVersion API is defined in the parent class VersionControl.
      boolean isBadVersion(int version); */

public class Solution extends VersionControl {
    public int firstBadVersion(int n) {
        int l = 0, r = n-1;
        while (l <= r) {
            int m = l + (r-l)/2;
            if (isBadVersion(m)) {
                r = m - 1;
            } else {
                l = m + 1;
            }
        }
        return l;
    }
}

File: 278-first-bad-version/278-first-bad-version.kt
/* The isBadVersion API is defined in the parent class VersionControl.
      fun isBadVersion(version: Int) : Boolean {} */

class Solution: VersionControl() {
    override fun firstBadVersion(n: Int) : Int {
        var l = 1
        var r = n
        var ans = -1
        while(l <= r) {
            var m = l + (r-l)/2
            if (isBadVersion(m)) {
                ans = m
                r = m - 1
            } else {
                l = m + 1
            }
        }
        return ans
	}
}